src/sentry/tasks/backfill_outboxes.py

```diff
@@ -82,108 +80,60 @@ def _chunk_processing_batch(
     return BackfillBatch(low=lower, up=capped, version=version, has_more=upper > capped)
 
 
-@instrumented_task(
-    name="sentry.tasks.backfill_outboxes.schedule_backfill_outbox_jobs_control",
-    queue="outbox.control",
-    acks_late=True,
-    silo_mode=SiloMode.CONTROL,
-)
-def schedule_backfill_outbox_jobs_control():
-    _schedule_backfill_outboxes(ControlOutboxProducingModel, process_backfill_outboxes_control)
+def process_outbox_backfill_batch(model: Type[Model], batch_size: int) -> BackfillBatch | None:
+    if not issubclass(model, RegionOutboxProducingModel) and not issubclass(
+        model, ControlOutboxProducingModel
+    ):
+        return None
+
+    processing_state = _chunk_processing_batch(model, batch_size=batch_size)
+    if not processing_state:
+        return None
+
+    for inst in model.objects.filter(id__gt=processing_state.low, id__lte=processing_state.up):
+        with outbox_context(transaction.atomic(router.db_for_write(model)), flush=False):
+            if isinstance(inst, RegionOutboxProducingModel):
+                inst.outbox_for_update().save()
+            if isinstance(inst, ControlOutboxProducingModel):
+                for outbox in inst.outboxes_for_update():
+                    outbox.save()
+
+    if not processing_state.has_more:
+        set_processing_state(model._meta.db_table, 0, model.replication_version + 1)
+    else:
+        set_processing_state(model._meta.db_table, processing_state.up, processing_state.version)
 
+    return processing_state
 
-@instrumented_task(
-    name="sentry.tasks.backfill_outboxes.schedule_backfill_outbox_jobs",
-    acks_late=True,
-    silo_mode=SiloMode.REGION,
-)
-def schedule_backfill_outbox_jobs():
-    _schedule_backfill_outboxes(RegionOutboxProducingModel, process_backfill_outboxes)
 
+OUTBOX_BACKFILLS_PER_MINUTE = 10_000
+
+
+def backfill_outboxes_for(
+    silo_mode: SiloMode, scheduled_count: int, max_batch_rate: int = OUTBOX_BACKFILLS_PER_MINUTE
+) -> bool:
+    # Maintain a steady state of outbox processing by subtracting any regularly scheduled rows
+    # from an expected rate.
+    remaining_to_schedule = max_batch_rate - scheduled_count
+    if remaining_to_schedule <= 0:
+        return False
```

Comment:
Should we have a metric/log message for this state? If an outbox shard is busy we might not be able to make much progress on a backfill as the regularly scheduled work could fill the 10000 records per minute threshold.

Reply:
That's fair, there is a `gauge` on how far backfills process, so it's possible to see the rate of all backfills and know when when it is complete / how much more remains.

Reply:
That works too. We also have the counters for new outboxes and processed ones.